geoengine/layers.py

```diff
@@ -486,16 +501,31 @@ def layer_collection(layer_collection_id: Optional[LayerCollectionId] = None,
     request = '/layers/collections' if layer_collection_id is None \
         else f'/layers/collections/{layer_provider_id}/{layer_collection_id}'
 
-    response = req.get(
-        f'{session.server_url}{request}?offset={offset}&limit={limit}',
-        headers=session.admin_or_normal_auth_header,
-        timeout=timeout,
-    )
+    page_limit = 20
+    pages: List[LayerCollectionResponse] = []
 
-    if not response.ok:
-        raise GeoEngineException(response.json())
+    offset = 0
+    while True:
+        response = req.get(
+            f'{session.server_url}{request}?offset={offset}&limit={page_limit}',
+            headers=session.admin_or_normal_auth_header,
+            timeout=timeout,
+        )
 
-    return LayerCollection.from_response(response.json())
+        if not response.ok:
+            raise GeoEngineException(response.json())
+
+        page: LayerCollectionResponse = response.json()
+
+        if len(page['items']) == 0:
```

Comment:
you can already stop requesting once fewer than `limit` elements are returned

Reply:
[c6694f3](https://github.com/geo-engine/geoengine-python/pull/83/commits/c6694f39632bea3bcd78a755615d28851e37daf9)